File: app/src/main/java/com/dream/countryinfo/CountryApp.java
package com.dream.countryinfo;

import android.app.Application;

import com.mapbox.mapboxsdk.Mapbox;

/**
 * Created by lixingming on 09/11/2017.
 */

public class CountryApp extends Application {

    private static CountryApp instance;

    @Override
    public void onCreate() {
        super.onCreate();

        instance = this;

        Mapbox.getInstance(getApplicationContext(), getResources().getString(R.string.mapbox_sdk_token));
    }

    public static CountryApp getApplication() {
        return instance;
    }

}

File: app/src/main/java/com/dream/countryinfo/CountryApp.kt
package com.dream.countryinfo

import android.app.Application

import com.mapbox.mapboxsdk.Mapbox

/**
 * Created by lixingming on 09/11/2017.
 */

class CountryApp : Application() {

    override fun onCreate() {
        super.onCreate()

        application = this

        Mapbox.getInstance(applicationContext, resources.getString(R.string.mapbox_sdk_token))
    }

    companion object {

        var application: CountryApp? = null
            private set
    }

}
